spec/jobs/assign_register_doi_job.rb

```diff
@@ -0,0 +1,45 @@
+require 'rails_helper'
+
+RSpec.describe AssignRegisterDoiJob do
+
+  let(:work) { FactoryBot.build(:dataset) }
+
+  describe '#perform' do
+    before do
+      allow(work).to receive(:reload) { work }
+    end
+    describe 'DOI assignment' do
+      before do
+        allow(CrossrefRegistration).to receive(:call).with(work)
+      end
+      it 'assigns a DOI to the work' do
+        expect(AssignDoi).to receive(:call).with(work)
+        subject.perform(work)
```

Comment:
Actually, I disagree.  It is not the responsibility of this class to set the DOI on the work.  That is the responsibility of the `AssignDoi` class.  It is the responsibility of this class to call the `AssignDoi` class with the proper argument(s) and that is what the test as written does.  So, I'm -1 on this suggested change ... though I'm willing to be convinced otherwise.